packages/block-library/src/footnotes/format.js

```diff
@@ -104,14 +146,10 @@ export const format = {
 				// When there is no footnotes block in the post, create one and
 				// insert it at the bottom.
 				if ( ! fnBlock ) {
-					const clientId = getSelectedBlockClientId();
-					let rootClientId = getBlockRootClientId( clientId );
-
-					while (
-						rootClientId &&
-						getBlockName( rootClientId ) !== 'core/post-content'
-					) {
-						rootClientId = getBlockRootClientId( rootClientId );
+					let rootClientId = getBlockRootClientId( selectedClientId );
+
+					if ( parentPostContent.length ) {
+						rootClientId = parentPostContent[ 0 ];
```

Comment:
I think the failing test in footnotes.spec.js is related to this change? (no longer using `while`) 

I might have to revert and do a follow up if so.

Reply:
yep, going to have to revert. I think it's related to nested blocks. 
Maybe we can tidy up in a follow up.